pyvista/core/grid.py

```diff
@@ -256,31 +256,53 @@ class UniformGrid(_vtk.vtkImageData, Grid, UniformGridFilters):
     - Initialize from a vtk.vtkImageData object
     - Initialize directly from the point arrays
 
-    See ``_from_specs`` in the documentation for more details on initializing
-    from point arrays
+    See ``_from_specs`` in the documentation for more details on
+    initializing from point arrays.
+
+    Parameters
+    ----------
+    uinput : str, vtk.DataSet, or pyvista.DataSet, optional
```

Comment:
The problem in this class init is that these parameters do not actually exist.  It is only by number of args.  kwargs isn't supported.

Reply:
An example of how this is confusing:`pyvista.UniformGrid(dims=(10, 10, 10))` creates a uniform grid with no cells or points.  Also sometimes dims can be the first arg depending on type and size.

Reply:
Good point, and I agree that this class init needs rework.

At the risk of expanding the scope of this PR, this has been fixed in 71d4ef69f40d9afc539df7e1e9427a80174c13f4.